.github/actions/detect-workflow/main.go

```diff
@@ -16,23 +16,57 @@ package main
 
 import (
 	"context"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"log"
+	"os"
 	"path"
 	"strings"
 
 	"github.com/slsa-framework/slsa-github-generator/github"
 )
 
-// getStr returns a string value from the given Event map. Values are specified
+type action struct {
+	getenv func(string) string
+	event  map[string]any
+	client *github.OIDCClient
+}
+
+// TODO(github.com/slsa-framework/slsa-github-generator/issues/153): use go-githubactions
```

Comment:
why do we need an external library for this?

Reply:
I'll leave out the TODO but we can discuss on #153